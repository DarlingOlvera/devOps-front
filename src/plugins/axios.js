import axios from 'axios'

//prueba local
const baseURL = 'http://localhost:9000/uadyfon/api/'

axios.defaults.baseURL = baseURL

axios.defaults.headers.common['Content-type'] = 'application/json'

let token = localStorage.getItem('tokenAuth')

if(token){
  axios.defaults.headers.common['Authorization'] = `Bearer${token}`
}
