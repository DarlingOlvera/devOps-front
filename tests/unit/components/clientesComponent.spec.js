import { shallowMount } from "@vue/test-utils";
import clientes from '@/components/clientesComponent.vue'
import axios from 'axios'

jest.mock('vuetify', () => ({
  useDisplay: jest.fn(() => ({ xs: true })),
}));

jest.mock('vue-router',()=>({
  useRouter: jest.fn(()=>({router:'login'}))
}))

const mockClientes = [
  {
    id:1,
    nombre:'Bob Esponja',
    edad:18,
    telefono:'999134730',
    tipo_compra:'equipo android',
    fecha_compra:'2023-05-12'
  },

]

jest.spyOn(axios, 'get').mockResolvedValue(mockClientes)

describe('Pruebas en clientesComponent', () => {

  test('should match snaptshot', () => {
    const wrapper = shallowMount(clientes)

    //expect(wrapper.html()).toMatchSnapshot()
  });

  test('Get telefonos debe devolver un arreglo de telefonos', async() => {
    const wrapper = shallowMount(clientes)

    expect(axios.get).toHaveBeenCalled()

    expect(axios.get).toHaveBeenCalledWith('clientes')

  });

});
