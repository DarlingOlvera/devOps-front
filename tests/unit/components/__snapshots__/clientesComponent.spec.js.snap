// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en clientesComponent should match snaptshot 1`] = `
<v-container fluid="" class="pa-10">
  <v-row class="d-flex justify-center">
    <h3 class="text-h4 text-title my-10 text-center">Información de clientes</h3>
  </v-row>
  <!--info moviles-->
  <v-row class="d-flex justify-center">
    <v-expansion-panels variant="accordion"></v-expansion-panels>
  </v-row>
  <vlayoutitem model-value="" position="bottom" class="text-end" size="88">
    <div class="ma-4">
      <vbtn icon="fa fa-solid fa-plus" size="large" color="primary" elevation="8"></vbtn>
    </div>
  </vlayoutitem>
</v-container>
<v-dialog modelvalue="false">
  <v-card width="100%" max-width="500" class="mx-auto pa-5" height="auto">
    <v-row class="d-flex justify-center">
      <h6 class="text-h6 text-title">Llene los datos</h6>
    </v-row>
    <v-row class="d-flex justify-center">
      <div class="pa-0" style="width: 80%;"><span class="text-caption text-title">Nombre</span>
        <v-text-field variant="outlined" modelvalue="" color="secondary" density="comfortable"></v-text-field>
      </div>
    </v-row>
    <v-row class="d-flex justify-center">
      <div class="pa-0" style="width: 80%;"><span class="text-caption text-title">Edad</span>
        <v-text-field variant="outlined" color="secondary" density="comfortable"></v-text-field>
      </div>
    </v-row>
    <v-row class="d-flex justify-center">
      <div class="pa-0" style="width: 80%;"><span class="text-caption text-title">telefono</span>
        <v-text-field variant="outlined" modelvalue="" color="secondary" density="comfortable"></v-text-field>
      </div>
    </v-row>
    <v-row class="d-flex justify-center">
      <div class="pa-0" style="width: 80%;"><span class="text-caption text-title">Tipo de compra</span>
        <v-text-field variant="outlined" modelvalue="" color="secondary" density="comfortable"></v-text-field>
      </div>
    </v-row>
    <v-row class="d-flex justify-center">
      <div class="pa-0" style="width: 80%;"><span class="text-caption text-title">Fecha de compra</span>
        <v-text-field variant="outlined" color="secondary" density="comfortable"></v-text-field>
      </div>
    </v-row>
    <v-row class="d-flex justify-center mb-5">
      <!--v-if-->
    </v-row>
  </v-card>
</v-dialog>
<v-dialog modelvalue="false">
  <v-card width="100%" max-width="500" class="mx-auto pa-5" height="auto">
    <v-row class="d-flex justify-center">
      <h6 class="text-h6 text-title">¿Está seguro?</h6>
    </v-row>
    <v-row class="d-flex justify-center mb-5">
      <v-btn color="red" class="mx-2">Sí</v-btn>
      <v-btn color="primary" class="mx-2">No</v-btn>
    </v-row>
  </v-card>
</v-dialog>
`;
